src/services/api/contract_class.rs

```diff
@@ -93,3 +112,100 @@ impl ContractClass {
         Ok(())
     }
 }
+
+// -------------------
+//  Helper Functions
+// -------------------
+
+impl TryFrom<PathBuf> for ContractClass {
+    type Error = io::Error;
+
+    fn try_from(path: PathBuf) -> io::Result<Self> {
```

Comment:
I'm pretty sure this could be `&Path`, or even `impl AsRef<Path>`.

Reply:
i tried to use path but the trait was not satisfied if i used it.

Reply:
Then maybe you could use a method since we're not actually converting anything from A to B.